Apps/Src/Task_CriticalState.c

```diff
@@ -1,26 +1,28 @@
 #include "os.h"
 #include "config.h"
+#include "BSP_Contactor.h"
+#include "CANbus.h"
 #include "Tasks.h"
 
 void Task_CriticalState(void *p_arg) {
     (void)p_arg;
 
     OS_ERR err;
     CPU_TS ts;
-
+    CANData_t Can_Data;
+    CANPayload_t Can_MSG; 
+    Can_MSG.idx = 0;
+    Can_Data.b = 1;
+    Can_Data.f = 0;
+    Can_Data.h = 0;
+    Can_Data.w = 0;
     // BLOCKING =====================
     // Wait until voltage, open wire, temperature, and current(Amperes) are all checked and safe
-    for(int32_t check = 0; check < NUM_FAULT_POINTS; check++) {
-        OSSemPend(&SafetyCheck_Sem4,
-                    0,
-                    OS_OPT_PEND_BLOCKING,
-                    &ts,
-                    &err);
-    }
-
+    while(OSSemPend(&SafetyCheck_Sem4, 0, OS_OPT_PEND_BLOCKING, &ts, &err) != 4);
     // Turn Contactor On
-    
+    BSP_Contactor_On();
     // Push All Clear message to CAN Q
-
+    CANbus_Send(ALL_CLEAR, Can_MSG);
     // Push Contactor State message to CAN Q
+    CANbus_Send(CONTACTOR_STATE, Can_MSG);
```

Comment:
With the way the CANBusConsumer task is planned out, the CAN messages should be pushed onto the CAN Bus queue which the CANBusCOnsumer task will process and call CANbus_Send. The CANBusConsumer task was added because multiple threads will be calling the CAN bus functions so the the queue will let the threads send CAN messages without blocking itself

Reply:
Not sure if the CANBus Consumer task has been implemented or not

Reply:
Should the pushing of the message into the queue be implemented in canbus.h or somewhere else?